apps/frontend/src/routes/chat/+page.server.ts

```diff
@@ -1,39 +1,24 @@
 import { error } from '@sveltejs/kit';
 import type { PageServerLoad } from './$types';
-import { createLangGraphClient } from '$lib/langgraph/client';
-
-async function initializeLangGraph(accessToken: string): Promise<{
-	threadId: string;
-	assistantId: string;
-}> {
-	const client = createLangGraphClient(accessToken);
-
-	const thread = await client.threads.create();
-	const assistant = await client.assistants.create({ graphId: 'chat' });
-
-	return {
-		threadId: thread.thread_id,
-		assistantId: assistant.assistant_id
-	};
-}
+import { createThread } from '$lib/langgraph/client';
 
 export const load: PageServerLoad = async ({ locals }) => {
 	const session = await locals.auth();
 
-	// If user is not authenticated, skip LangGraph setup.
-	// Returning `session: null` allows client to show login modal instead of erroring.
 	if (!session?.accessToken) {
-		return { session: null };
+		return { session };
 	}
 
 	try {
-		const { threadId, assistantId } = await initializeLangGraph(session.accessToken);
+		// Create a LangGraph client, thread, and assistant
+		// Pre-creates a LangGraph thread and assistant on page load to improve perceived performance.
+		// This is a tradeoff: it enables instant chat start, but creates a thread even if the user doesn't chat.
+		// Revisit if lazy-init becomes more appropriate.
+		const langgraph = await createThread(session.accessToken);
+
```

Comment:
There's something to say for not doing any langgraph interactions server-side. It's a bit slower but it's a bit more strict in terms of separation of concerns. Note that even if we do it client-side, we can do the init async as soon as the window loads so that by the time the user wants to start chatting things are already nicely set up.

But let's keep this for now.